src/pages/deal/ResourceTable.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react'
+
+import {
+  Cell,
+  Row,
+  RowBlock,
+  RowHeader,
+  RowTrigger,
+  ScrollableTable,
+  TableBody,
+  TableColumnTitle,
+  TableHeader,
+} from '../../components/Table'
+import { Text } from '../../components/Text'
+import { JsonToYamlView } from '../../components/YamlView'
+import { getResourceName } from '../../utils/getResourceName'
+import { formatHexData } from '../../utils/helpers'
+
+const template = [
+  'minmax(10px, 1fr)',
+  'minmax(10px, 1fr)',
+  'minmax(10px, 1fr)',
+  'minmax(10px, 1fr)',
+  'minmax(200px, 1fr)',
+]
+
+interface ResourceTableProps {
+  resources?:
+    | {
+        id: string
+        details?: string
+        quantity: string
+        metadata?: string
+        type: number
+      }[]
+    | null
+}
+
+export const ResourceTable: React.FC<ResourceTableProps> = ({ resources }) => {
+  return (
+    <ScrollableTable>
+      <TableHeader template={template}>
+        <TableColumnTitle>Resource ID</TableColumnTitle>
+        <TableColumnTitle>Type</TableColumnTitle>
+        <TableColumnTitle>Requested</TableColumnTitle>
+        <TableColumnTitle>Metadata</TableColumnTitle>
+      </TableHeader>
+      <TableBody>
+        {resources?.map((resource) => (
+          <ResourceRow key={resource.id} resource={resource} />
+        ))}
+      </TableBody>
+    </ScrollableTable>
+  )
+}
+
+interface ResourceRowProps {
+  resource: {
+    id: string
+    details?: string
+    quantity: string
+    metadata?: string
```

Comment:
I am curious why metadata is optional. All resources have metadata right?

Reply:
We have some problems with resolving ResourceDescription data in ResourceData but fixed that in the Indexer. I will update ts-client little bit later

Reply:
are there still problems?